main.go

```diff
@@ -254,30 +309,60 @@ func _main() error {
 	if err != nil {
 		return err
 	}
+	projectID := projectTmp.Name()
+	if cache {
+		stdinTmpCopy := false
+		projectID, stdinTmpCopy, err = getFilesContentHash(files, projectTmp.Name())
+		if err != nil {
+			fmt.Printf("Error creating hash for cache mode: %v, defaulting to normal mode", err)
+			cache = false
+		}
+		if stdinTmpCopy {
+			files = append(files, projectTmp.Name())
+		}
+	}
 	defer os.Remove(projectTmp.Name())
+
 	project := &runner.ProjectState{
-		Id: projectTmp.Name(),
+		Id: projectID,
 		Pages: []runner.ProjectPage{
 			{
 				Panels: nil,
 			},
 		},
 	}
 
+	settings := *runner.DefaultSettings
 	tmpDir, err := os.MkdirTemp("", "dsq")
 	if err != nil {
 		return err
 	}
 	defer os.RemoveAll(tmpDir)
+	if cache {
+		tmpDir = "dsq-cache"
+		settings.CacheMode = &cache
+	}
+	ec := runner.NewEvalContext(settings, tmpDir)
 
-	ec := runner.NewEvalContext(*runner.DefaultSettings, tmpDir)
 	for i := 0; i < len(files); i++ {
```

Comment:
I think that when we're in cache mode and the cache is valid, this loop shouldn't run at all.

The only reason we run it is to grab shapes because the evaldatabase call below expects the project to have valid shapes.

But when we've already cached we don't need to check shapes. Shapes are only used during imports, not during query.

So I think we should find a way to skip this loop completely when the cache is valid.

Reply:
hmm okay, that means we don't have to save all the json files too.

Reply:
Right! The only thing that needs to stick around is the sqlite file.